presto-main/src/main/java/com/facebook/presto/operator/DriverContext.java

```diff
@@ -159,24 +159,23 @@ public void recordBlocked(ListenableFuture<?> blocked)
 
     public void finished()
     {
+        freeMemory(memoryReservation.get());
```

Comment:
Shouldn't this be done after the state change? Otherwise, if this method is called twice, we'll free memory twice (and the intention seems to be that multiple calls is a no-op)